src/otx/core/data/pre_filtering.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Pre filtering data for OTX."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from datumaro.components.annotation import Annotation, Bbox, Polygon
+from datumaro.components.dataset import Dataset as DmDataset
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset_base import DatasetItem
+
+
+def pre_filtering(dataset: DmDataset) -> DmDataset:
+    """Filtering invalid data in datumaro dataset."""
+    dataset = DmDataset.filter(dataset, is_non_empty_item)
+    dataset = DmDataset.filter(dataset, is_valid_annot, filter_annotations=True)
+
+    return remove_unused_labels(dataset)
+
+
+def is_non_empty_item(item: DatasetItem) -> bool:
+    """Return whether DatasetItem's annotation is non-empty."""
+    return not (item.subset == "train" and len(item.annotations) == 0)
+
+
+def is_valid_annot(item: DatasetItem, annotation: Annotation) -> bool:
+    """Return whether DatasetItem's annotation is valid."""
+    del item
+    if isinstance(annotation, Bbox):
+        x1, y1, x2, y2 = annotation.points
+        return x1 < x2 and y1 < y2
```

Comment:
```suggestion
    if isinstance(annotation, Bbox):
        msg = "There are bounding box which is not `x1 < x2 and y1 < y2`. They will be filtered out before training."
        warnings.warn(msg)
        x1, y1, x2, y2 = annotation.points
        return x1 < x2 and y1 < y2
```
Please add warning to prevent an user dataset from being tampered with without warning.

Reply:
Done by 
7689029

Reply:
> Please add this warning to others.
Did you insert warnings for all other cases that can modify the given dataset?

Reply:
Done by [8b3625d](https://github.com/openvinotoolkit/training_extensions/pull/2823/commits/8b3625d338c2296e9f4a3399877d25d2430c8e25)